pkg/database/realm.go

```diff
@@ -1287,12 +1334,28 @@ func (r *Realm) SigningKeyID() string {
 	return fmt.Sprintf("realm-%d", r.ID)
 }
 
+func (r *Realm) SMSSigningKeyID() string {
+	return fmt.Sprintf("realm-sms-%d", r.ID)
+}
+
 // CreateSigningKeyVersion creates a new signing key version on the key manager
 // and saves a reference to the new key version in the database. If creating the
 // key in the key manager fails, the database is not updated. However, if
 // updating the signing key in the database fails, the key is NOT deleted from
 // the key manager.
 func (r *Realm) CreateSigningKeyVersion(ctx context.Context, db *Database) (string, error) {
+	newKey := &SigningKey{}
+	return r.createdManagedSigningKey(ctx, db, r.SigningKeyID(), "signing_keys", "certificate", newKey)
+}
+
+// CreateSMSSigningKeyVersion creates a new SMS signing key versino on the key manager
+// and saves a reference to the new key version in the database.
+func (r *Realm) CreateSMSSigningKeyVersion(ctx context.Context, db *Database) (string, error) {
+	newKey := &SMSSigningKey{}
+	return r.createdManagedSigningKey(ctx, db, r.SMSSigningKeyID(), "sms_signing_keys", "SMS", newKey)
+}
+
+func (r *Realm) createdManagedSigningKey(ctx context.Context, db *Database, keyID, table, purpose string, signingKey ManagedKey) (string, error) {
```

Comment:
Instead of passing in `purpose`, you can use `%T` on the `signingKey` param in the errors. It seems that's the only purpose for passing in purpose.

Reply:
I moved Table() and Purpose() to methods on the ManagedKey interface.